### PR TITLE
Added the features per issue 66

### PR DESCRIPTION
**Pull request by n1lsw**

### CODE REVIEW

File: src/pages/api/autocomplete.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { z, ZodError } from "zod";
+import { MasterItem, List, Prisma, PrismaClient } from "@prisma/client";
+var stringSimilarity = require("string-similarity");
```
Comment (NoHara42) [inline src/pages/api/autocomplete.ts:4]: this needs to be added as a dependency to the package.json file

File: src/pages/api/autocomplete.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    }
+    if (a.rating > b.rating) {
+      return -1;
+    }
+    return 0;
+  });
+  return sorted;
+}
+
+const prisma = new PrismaClient();
```
Comment (NoHara42) [inline src/pages/api/autocomplete.ts:21]: this should be instantiated only once, at a higher level of the application and shared via exporting the constant

read more here:
https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/instantiate-prisma-client#the-number-of-prismaclient-instances-matters

File: src/pages/api/autocomplete.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { z, ZodError } from "zod";
+import { MasterItem, List, Prisma, PrismaClient } from "@prisma/client";
+var stringSimilarity = require("string-similarity");
+
+export function sortByRating(input: Array<{ target: string; rating: number }>) {
+  const sorted = [...input];
```
Comment (NoHara42) [inline src/pages/api/autocomplete.ts:7]: this is the same as writing `const sorted = input;`

Comment (NoHara42) [reply src/pages/api/autocomplete.ts:7]: or are you deep-copying here?

Comment (n1lsw) [reply src/pages/api/autocomplete.ts:7]: We want to create a copy of the input array.

File: src/pages/api/autocomplete.ts (hunk 2 above)
Comment (n1lsw) [reply src/pages/api/autocomplete.ts:21]: Created an instance of the prisma client on the top level index.ts

File: src/pages/api/autocomplete.ts (hunk 1 above)
Comment (n1lsw) [reply src/pages/api/autocomplete.ts:4]: Added the dependency

**Review by NoHara42 — APPROVED** [review]
approving so that this can be tested with a later PR that contains the schema changes

**Pull request merged**